hardware_frame_context.go

```diff
@@ -0,0 +1,48 @@
+package astiav
+
+//#include <libavcodec/avcodec.h>
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+type HardwareFrameContext struct {
+	c *C.struct_AVBufferRef
+}
+
+func AllocHardwareFrameContext(hdc *HardwareDeviceContext) (*HardwareFrameContext, error) {
```

Comment:
Can you not return an error here but only a `*HardwareFrameContext` and let the developer checks whether it's nil or not?